Source/WebCore/rendering/updating/RenderTreeBuilderList.cpp

```diff
@@ -49,6 +49,7 @@ static bool generatesLineBoxesForInlineChild(RenderBlock& current, RenderObject*
 static RenderBlock* getParentOfFirstLineBox(RenderBlock& current, const RenderObject& marker)
```

Comment:
I recently landed a few PRs around this code so I think now you can move this logic to `parentCandidateForMarker` something along the lines of

 ```
   if (marker.isInside()) {
        if (firstChild->isBlockLevelBox())
          return &listItemRenderer;
       return findParentOfEmptyOrFirstLineBox(listItemRenderer, marker).first;
    }

```

(with some function signature changes (`RenderBlock& blockContainer` -> `RenderListItem& listItemRenderer` and grabbing the first child as RenderElement)

or you can simply land your change and I'll move it over to `parentCandidateForMarker`. I am fine with either.